src/utils/formatNumbers.ts

```diff
@@ -569,18 +581,47 @@ export function useFormatterLocales(): {
 
 // used to get the appropriate formatter functions with the correct locales passed into them
 export function useFormatter() {
+  const activeLocalCurrency = useActiveLocalCurrency()
   const { formatterLocale, formatterLocalCurrency } = useFormatterLocales()
 
+  const activeLocalCurrencyIsUSD = activeLocalCurrency === GqlCurrency.Usd
+  const { data: localCurrencyConversionRate, isLoading: localCurrencyConversionRateIsLoading } =
+    useLocalCurrencyConversionRate(activeLocalCurrency, activeLocalCurrencyIsUSD)
+
+  const previousSelectedCurrency = usePrevious(activeLocalCurrency)
+  const previousConversionRate = usePrevious(localCurrencyConversionRate)
+
+  const shouldFallbackToPrevious = !localCurrencyConversionRate && localCurrencyConversionRateIsLoading
+  const shouldFallbackToUSD = !localCurrencyConversionRate && !localCurrencyConversionRateIsLoading
+  const currencyToFormatWith = shouldFallbackToUSD
+    ? DEFAULT_LOCAL_CURRENCY
+    : shouldFallbackToPrevious
+    ? previousSelectedCurrency
+    : formatterLocalCurrency
```

Comment:
nit: prefer not nesting ternary logic

Reply:
I also agree with you - I too hate the ternaries, however they let me declare this as a const so to me that was more upside than a mutable variable